core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -214,7 +214,10 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         }
 
         // check if new domain is wider than domain already provided by table scan
-        if (constraint.predicate().isEmpty() && newDomain.contains(node.getEnforcedConstraint())) {
+        if (constraint.predicate().isEmpty() &&
+                // TODO do we need to track enforced ConnectorExpression in TableScanNode?
```

Comment:
Do you mean that currently any expression that gets pushed down is not accounted for in the TableScanNode? Doesn't this also mean that the explain output would be missing some constraints that were actually applied?

Reply:
> Doesn't this also mean that the explain output would be missing some constraints that were actually applied?

This is `ConnectorTableHandle.toString` responsibility

> Do you mean that currently any expression that gets pushed down is not accounted for in the TableScanNode?

Currently, yes. This is hard problem though, because what we want is to track `io.trino.sql.planner.plan.TableScanNode#enforcedConstraint` (but for expression), and we need an equivalent of `io.trino.spi.predicate.TupleDomain#overlaps` for expressions.